extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -185,16 +193,30 @@ export class SubscriptionService {
 
   /**
    * @param {!SubscriptionPlatform} subscriptionPlatform
-   * @return {!Promise<!./entitlement.Entitlement>}
+   * @return {!Promise}
    */
   fetchEntitlements_(subscriptionPlatform) {
-    return subscriptionPlatform.getEntitlements().then(entitlement => {
-      if (!entitlement) {
-        entitlement = Entitlement.empty(subscriptionPlatform.getServiceId());
-      }
+    let timeout = SERVICE_TIMEOUT;
+    if (getMode().development || getMode().localDev || getMode().log) {
+      timeout = SERVICE_TIMEOUT * 2;
+    }
+    return this.timer_.timeoutPromise(
+        timeout,
+        subscriptionPlatform.getEntitlements()
+    ).then(entitlement => {
+      entitlement = entitlement || Entitlement.empty(
+          subscriptionPlatform.getServiceId());
       this.resolveEntitlementsToStore_(subscriptionPlatform.getServiceId(),
           entitlement);
       return entitlement;
+    }).catch(reason => {
+      const serviceId = subscriptionPlatform.getServiceId();
+      this.platformStore_.reportPlatformFailure(serviceId);
+      this.resolveEntitlementsToStore_(serviceId,
+          Entitlement.empty(serviceId));
+      user().createError(
```

Comment:
You still have to call `throw`. E.g.
```
throw user().createError(...)
```

Reply:
I am a little confused here, I though its just for logging. Which it does pretty well.
Moreover all the usage in the AMP repo for `user().createError` is for logging. Any reason why we want to throw as well?